### PR TITLE
Polish for logs view

### PR DESCRIPTION
**Pull request by ccschmitz**
## Summary

* Some UI updates to better match what we have in Figma
* Adds a copy button to copy the raw log to your clipboard
* Adds a button to toggle all nested fields
* Adds some logic to collapse all rows when the search query changes
* Adds a `kind` prop to `ButtonLink` that allows creating button links that are gray instead of purple
* Fixes a display issue when logs have very long string values with no spaces

https://www.loom.com/share/a744874144484127845f78a4eab3d649

## How did you test this change?

Local and PR preview click test

## Are there any deployment considerations?

N/A

### CODE REVIEW

**Review by Vadman97 — APPROVED** [review]

**Review by et — APPROVED** [review]
Beautiful!

**Comment by Vadman97** [comment]
@ccschmitz bit of a bug with the nested keys rendering?
https://highlightazureexample.azurewebsites.net/api/HttpTrigger?name=jared
Screenshot 2023-02-24 at 3 50 29 PM

**Comment by ccschmitz** [comment]
@Vadman97 is `0` and integer or a string? I think we need to handle display of numbers. Will fix!

File: frontend/src/pages/LogsPage/LogsTable/LogsTable.tsx [hunk 1]
```diff
@@ -136,6 +104,11 @@ const LogsTable = ({ data, loading, query }: Props) => {
 		debugTable: true,
 	})
 
+	useEffect(() => {
+		// Collapse all rows when search changes
+		table.toggleAllRowsExpanded(false)
+	}, [logs])
```
[the remark this answers is not in the document]
Comment (et) [reply frontend/src/pages/LogsPage/LogsTable/LogsTable.tsx:110]: Tracked in #4404

[the remark this answers is not in the document]
Comment (ccschmitz) [reply frontend/src/pages/LogsPage/LogsTable/LogsTable.tsx:110]: Yeah, I think this was a mistake to pass an array to the dependency list. We may want to use `loading` instead.

[the remark this answers is not in the document]
Comment (et) [reply frontend/src/pages/LogsPage/LogsTable/LogsTable.tsx:110]: @ccschmitz - I fixed the issue in #4424. Seems to working but let me know if you think it should be adjusted.

[the remark this answers is not in the document]
Comment (ccschmitz) [reply frontend/src/pages/LogsPage/LogsTable/LogsTable.tsx:110]: Oh, nice! Thank you!

**Pull request merged**